.github/workflows/test-kyuubi.yaml

```diff
@@ -0,0 +1,169 @@
+      - name: Clone repo
+        run: |
+          ssh-keyscan -t rsa git.launchpad.net >> ~/.ssh/known_hosts
+          echo "cloning..."
+          git clone --progress --verbose ${{ secrets.REPO_KYUUBI_URL }}
+          echo "cloned!"
+          cp -r ie-tests charmed-kyuubi
+      - name: Generate uuid
+        id: uuid
+        timeout-minutes: 5
+        run: |
+          # Generate uuid
+          uuid=$(uuidgen)
+          echo "uuid=$uuid" >> $GITHUB_OUTPUT
+
+      - name: Setup maven credential
+        run: |
+          echo "Setup custom settings.xml"
+          mkdir ~/.m2
+          mv ie-tests/settings.xml  ~/.m2/settings.xml        
+      
+      - name: Run integration test
+        timeout-minutes: 180
+        env:
+          USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+          PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+          ARTIFACTORY_URL: ${{ secrets.KYUUBI_ARTIFACTORY_URL }}
+          JAVA_HOME: /usr/lib/jvm/java-1.17.0-openjdk-amd64/
+          SPARK_LOCAL_IP: localhost
+          TEST_MODULES: dev/kyuubi-codecov
+        run: |
+          echo "Check Java version"
+          which java
+          # set java version to Java 17
+          sudo update-java-alternatives -s $(sudo update-java-alternatives -l | grep '17' | cut -d " " -f1) || echo '.'
+          echo "Java version"
+          java -version
+          echo "Maven version"
+          mvn -version
+
+          cd charmed-kyuubi
+
+
+          branch=${{ matrix.branch }}
+          echo "Branch: $branch" 
+          if [ -z "$branch" ];
+          then 
+            branch="lp-1.10.2"
+          fi
+          echo "Branch to test: $branch"         
+          # /bin/bash run.sh $branch
+
+          git checkout $branch
+
+          # create test folder
+          mkdir -p ie-tests/logs
+          
+          # add extra folder and files to be not scanned by rat
+          echo "ie-tests/*" >> .rat-excludes
+          echo "**/ie-tests/*" >> .rat-excludes
+          echo "lp/*" >> .rat-excludes
+          ls -larth
+          echo "Print rat-excludes"
+          cat .rat-excludes
+
+          FOLDER=$(pwd)
+          IE_FOLDER=$FOLDER/ie-tests
+          DATE=$(date +%Y%m%dH%HM%M)
+          BRANCH=$(git rev-parse --abbrev-ref HEAD)
+          COMMIT=$(git rev-parse --short HEAD)
+          LOG_NAME=${DATE}_${BRANCH}_${RANDOM}_test
+          STD_OUT="${IE_FOLDER}/logs/${LOG_NAME}.out"
+          STD_ERR="${IE_FOLDER}/logs/${LOG_NAME}.err"
+
+          echo "Start integration tests for Kyuubi"
+          mvn clean install -pl $TEST_MODULES -am -Pjava-17 -Pspark-3.5 -Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.4.3 -Dspark.archive.name=spark-3.4.3-bin-hadoop3.tgz -Pzookeeper-3.6 -Dmaven.plugin.scalatest.exclude.tags=org.scalatest.tags.Slow,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest,org.apache.kyuubi.tags.PaimonTest,org.apache.kyuubi.tags.SparkLocalClusterTest >> $STD_OUT 2> $STD_ERR 
```

Comment:
Oh my gosh. Figuring out this line might have been not that easy. 

The only small concern I would have here is that from this line is would seem that we exclude Delta and Iceberg tests? Is this correct? If so, why?

Reply:
I am following what upstream is doing.